students/cfarris/lessons/lesson05/assignment/test_database.py

```diff
@@ -0,0 +1,140 @@
+    """
+    Tests read csv function is able to pull in a csv and return a dictionary
+    That has the header for column as key and the value from row as value
+    """
+    result = read_csv('rentals.csv')
+    print(result)
+    assert result == [{'product_id': 'prd003', 'user_id': 'user004'},
+                      {'product_id': 'prd002', 'user_id': 'user008'},
+                      {'product_id': 'prd002', 'user_id': 'user005'},
+                      {'product_id': 'prd005', 'user_id': 'user001'},
+                      {'product_id': 'prd010', 'user_id': 'user002'},
+                      {'product_id': 'prd007', 'user_id': 'user002'},
+                      {'product_id': 'prd006', 'user_id': 'user003'},
+                      {'product_id': 'prd005', 'user_id': 'user003'},
+                      {'product_id': 'prd001', 'user_id': 'user010'}]
+
+
+
+
+def test_import_data(set_up_connection):
+    """
+    Tests main function which intakes a dict of names and csv files, and 
+    creates a collection based on value and adds documents from csv file to that
+    collection. 
+    This test tests that Main returns the three collections 
+    (products_db, rentals_db, customers_db)
+    """
+    db = set_up_connection
+    result = import_data(db, coll_dict)
+    print(result)
+    assert result == (9, 10, 10)
+
+
+def test_show_avail_products(set_up_connection):
+    """
+    tests whether show available products works. Is dependent on having a products_db
+    This te
+    """
+    db = set_up_connection
+    result = import_data(db, coll_dict)
+    results = show_avail_products(db)
+    print(results)
+    assert results == {'prd001': {'description': '60-inch TV stand',
+                                  'product_type': 'livingroom',
+                                  'quantity_available': '3'},
+                       'prd003': {'description': 'Acacia kitchen table',
+                                  'product_type': 'kitchen',
+                                  'quantity_available': '7'},
+                       'prd004': {'description': 'Queen bed',
+                                  'product_type': 'bedroom',
+                                  'quantity_available': '10'},
+                       'prd005': {'description': 'Reading lamp',
+                                  'product_type': 'bedroom',
+                                  'quantity_available': '20'},
+                       'prd006': {'description': 'Portable heater',
+                                  'product_type': 'bathroom',
+                                  'quantity_available': '14'},
+                       'prd008': {'description': 'Smart microwave',
+                                  'product_type': 'kitchen',
+                                  'quantity_available': '30'},
+                       'prd010': {'description': '60-inch TV',
+                                  'product_type': 'livingroom',
+                                  'quantity_available': '3'}}
```

Comment:
This is starting to get unwieldy. Think about how the information can be available for both the tests and the code you are testing.